src/pal/src/file/file.cpp

```diff
@@ -1136,14 +1168,17 @@ DeleteFileA(
     ENTRY("DeleteFileA(lpFileName=%p (%s))\n", lpFileName?lpFileName:"NULL", lpFileName?lpFileName:"NULL");
 
     pThread = InternalGetCurrentThread();
-    if (strlen(lpFileName) >= MAX_LONGPATH)
+    length = strlen(lpFileName);
+    if (length >= MAX_LONGPATH)
```

Comment:
We still want to keep the check here to throw error as OS doesn't support path lengths greater than MAX_LONGPATH. Allowing error handling to happen at OS level, and letting OS throw if the path lengths are greater than supported max is a work in backlog. We'll defer these changes to a future commit.